app/models/host/managed.rb

```diff
@@ -848,6 +848,26 @@ def setup_clone
     @old = super { |clone| clone.interfaces = self.interfaces.map {|i| setup_object_clone(i) } }
   end
 
+  # rebuilds orchestration configuration for a host
+  # takes all the methods from Orchestration modules that are registered for configuration rebuild
+  # returns  : Hash with 'true' if rebuild was a success for a given key (Example: {"TFTP" => true, "DNS" => false})
+  def recreate_config
+    result = {}
+
+    Nic::Managed.rebuild_methods.map do |method, pretty_name|
+      interfaces.map do |interface|
+        value = interface.send method
+        result[pretty_name] = value if !result.has_key?(pretty_name) || (result[pretty_name] && !value)
+      end
+    end
+
+    self.class.rebuild_methods.map do |method, pretty_name|
+      fail "The name '#{pretty_name}' is already taken by some other orchestration module, please choose another name" if result[pretty_name]
```

Comment:
This could be problematic, as it will fail at runtime for the user while the other "fail" is at load time (so we'd spot it during tests).  I think it should raise a Foreman::Exception here.